drivers/regulator/regulator_cp9314.c

```diff
@@ -283,14 +269,75 @@ static int regulator_cp9314_get_error_flags(const struct device *dev,
 	return 0;
 }
 
+static int regulator_cp9314_write_lock(const struct device *dev,
+				       const enum cp9314_backdoor_keys key)
+{
+	const struct regulator_cp9314_config *config = dev->config;
+	struct regulator_cp9314_data *data = dev->data;
+	int ret;
+
+	if (data->allow_hw_i2c_lock == 0U) {
```

Comment:
It's a bool so treat is as such? :) prob apply elsewhere for hw_i2c_lock too

Reply:
Ack, here and elsewhere.